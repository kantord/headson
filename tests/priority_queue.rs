use insta::assert_snapshot;
use serde_json::Value;

#[test]
fn pq_empty_array() {
    let value: Value = serde_json::from_str("[]").unwrap();
    let pq = headson::build_priority_queue(&value).unwrap();
    let mut lines = vec![format!("len={}", pq.len())];
    for (item, prio) in pq.into_sorted_iter() {
        lines.push(format!("{:?} prio={}", item, prio));
    }
    assert_snapshot!("pq_empty_array_queue", lines.join("\n"));
}

#[test]
fn pq_single_string_array() {
    let value: Value = serde_json::from_str("[\"ab\"]").unwrap();
    let pq = headson::build_priority_queue(&value).unwrap();
    let mut lines = vec![format!("len={}", pq.len())];
    for (item, prio) in pq.into_sorted_iter() {
        lines.push(format!("{:?} prio={}", item, prio));
    }
    assert_snapshot!("pq_single_string_array_queue", lines.join("\n"));
}
