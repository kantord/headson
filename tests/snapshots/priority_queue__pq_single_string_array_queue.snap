---
source: tests/priority_queue.rs
expression: "lines.join(\"\\n\")"
---
len=4
QueueItem { node_id: NodeId(3), parent_id: ParentId(Some(1)), kind: String, depth: 2, index_in_array: Some(1), priority: 3, value_repr: "\"b\"" } prio=3
QueueItem { node_id: NodeId(2), parent_id: ParentId(Some(1)), kind: String, depth: 2, index_in_array: Some(0), priority: 2, value_repr: "\"a\"" } prio=2
QueueItem { node_id: NodeId(1), parent_id: ParentId(Some(0)), kind: String, depth: 1, index_in_array: Some(0), priority: 1, value_repr: "\"ab\"" } prio=1
QueueItem { node_id: NodeId(0), parent_id: ParentId(None), kind: Array, depth: 0, index_in_array: None, priority: 0, value_repr: "[\"ab\"]" } prio=0
